flake.nix

```diff
@@ -22,37 +32,79 @@
             openssl
             pkg-config
             perl
+            fenix-channel.rustc
           ];
 
           nativeBuildInputs = with pkgs; [
             pkg-config
           ];
         };
 
-        cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
-          pname = "minimint";
+        minimintDeps = craneLib.buildDepsOnly (commonArgs // {
+          pname = "minimint-dependencies";
         });
 
-        myCrateClippy = craneLib.cargoClippy (commonArgs // {
-          inherit cargoArtifacts;
+        minimintClippy = craneLib.cargoClippy (commonArgs // {
+          cargoArtifacts = minimintDeps;
           cargoClippyExtraArgs = "-- --deny warnings";
         });
 
-        myCrate = craneLib.buildPackage (commonArgs // {
-          inherit cargoArtifacts;
+        minimint = craneLib.buildPackage (commonArgs // {
+          cargoArtifacts = minimintDeps;
         });
 
-        myCrateCoverage = craneLib.cargoTarpaulin (commonArgs // {
-          inherit cargoArtifacts;
+        minimintCoverage = craneLib.cargoTarpaulin (commonArgs // {
+          cargoArtifacts = minimintDeps;
         });
       in
       {
-        packages.default = myCrate;
+        packages.default = minimint;
         checks = {
-         inherit
-           myCrate
-           myCrateClippy
-           myCrateCoverage;
+          inherit
+            minimint
+            minimintClippy
+            minimintCoverage;
         };
+
+        devShell =
+          let
+            clightning-dev = pkgs.clightning.overrideAttrs (oldAttrs: {
+              configureFlags = [ "--enable-developer" "--disable-valgrind" ];
+            } // pkgs.lib.optionalAttrs (!pkgs.stdenv.isDarwin) {
+              NIX_CFLAGS_COMPILE = "-Wno-stringop-truncation";
+            });
+            bitcoind-patch-darwin = pkgs.bitcoind.overrideAttrs (oldAttrs: {
+              doCheck = !(pkgs.stdenv.isDarwin && pkgs.stdenv.isAarch64);
+            });
+          in
+
+          pkgs.mkShell {
+            buildInputs = minimintDeps.buildInputs;
+            nativeBuildInputs = with pkgs; minimintDeps.nativeBuildInputs ++ [
+              fenix-pkgs.rust-analyzer
+              fenix-channel.rustfmt
+              fenix-channel.rustc
+              fenix-channel.cargo
+
+              bc
+              perl
+              bitcoind-patch-darwin
+              clightning-dev
+              jq
+              procps
+              tmux
+              tmuxinator
+
+              # Nix
```

Comment:
Couple of nice things, like all the relevant LSPs, and `nixpkg-fmt` for formatting nix. I like to make `nix develop` install all the git templates and checks that would use them. https://github.com/rustshop/rustshop/blob/a8ff84a8f3d8522fb218574cb9f6b7935596aceb/flake.nix#L192

Reply:
If you have time, I have a blogpost about what my aims are: https://github.com/rustshop/rustshop/discussions/17